tests/test_file_open_gui.py

```diff
@@ -0,0 +1,348 @@
+    ].exists(), "My Tasks Only checkbox is missing"
+    assert app_dialog.root.buttons[
+        "+ New Task"
+    ].exists(), "+ New Task button is missing"
+    assert app_dialog.root[
+        "Search Entity"
+    ].exists(), "Search Assets text field is missing"
+
+    # Got to the model task and validate breadcrumb
+    app_dialog.root.outlineitems["Character"].waitExist(timeout=30)
+    app_dialog.root.outlineitems["Character"].mouseDoubleClick()
+    app_dialog.root.outlineitems["AssetAutomation"].waitExist(timeout=30)
+    app_dialog.root.outlineitems["AssetAutomation"].mouseDoubleClick()
+    app_dialog.root.outlineitems["Model"].waitExist(timeout=30)
+
+    # Validate content dialog
+    assert app_dialog.root.cells[
+        "AssetAutomation"
+    ].exists(), "AssetAutomation is missing in content dialog"
+    assert app_dialog.root.cells[
+        "Model - Model"
+    ].exists(), "Model task is missing in content dialog"
+    assert app_dialog.root.cells[
+        "Rig - Rig"
+    ].exists(), "Rig task is missing in content dialog"
+
+    # Search in the content dialog for Rig and make sure Model is not showing up anymore
+    app_dialog.root["Search All Files"].typeIn("Rig" "{ENTER}")
+    assert app_dialog.root.cells[
+        "Rig - Rig"
+    ].exists(), "Rig task should be visible in content dialog"
+    assert (
+        app_dialog.root.cells["Model - Model"].exists() is False
+    ), "Model task shouldn't be visible in content dialog"
+
+    # Remove text in the search field and make sure Modal task is back
+    app_dialog.root["Search All Files"].buttons.mouseClick()
+    assert app_dialog.root.cells[
+        "Model - Model"
+    ].exists(), "Model task should be visible in content dialog"
+
+    # Select Model task
+    app_dialog.root.outlineitems["Model"].mouseDoubleClick()
+    app_dialog.root.outlineitems["Model"][1].waitExist(timeout=30)
+    app_dialog.root.outlineitems["Model"][1].mouseClick()
+
+    # Validate content dialog
+    assert app_dialog.root.cells["Model - Model"].exists(), "Not on the right tasks"
+
+    # Validate breadcrumb
+    assert app_dialog.root.captions[
+        "Assets * Character * Asset AssetAutomation * Step Model * Task Model"
+    ].exists(), "Breadcrumb not on the right task"
+
+    # Click on the back navigation button until back to the Assets context
+    for _i in range(0, 4):
+        # Click on the back navigation button
+        app_dialog.root.buttons["nav_prev_btn"].mouseClick()
+
+    # Validate Breadcrumb widget is only showing the project context
+    assert app_dialog.root.captions[
+        "Assets"
+    ].exists(), "Breadcrumb widget is not set correctly"
+
+    # Enable My Tasks Only and make sure Model task is not showing up anymore
+    app_dialog.root.checkboxes["My Tasks Only"].mouseClick()
+    assert (
+        app_dialog.root.outlineitems["Model"].exists() is False
+    ), "Model task shouldn't be visible"
+
+
+def test_shots_tab(app_dialog):
```

Comment:
It looks like the shot and asset tabs do the same thing really, just with Asset/Shot and different task names.
Perhaps this could be refactored into a single test_tab method, and parametrize like the Python console does here:
https://github.com/shotgunsoftware/tk-multi-pythonconsole/blob/master/tests/test_standalone_app.py#L137-L150